EngineLayer/CrosslinkSearch/CrosslinkSpectralMatch.cs

```diff
@@ -9,26 +9,35 @@ namespace EngineLayer.CrosslinkSearch
 {
     public class CrosslinkSpectralMatch : PeptideSpectralMatch
     {
-        public CrosslinkSpectralMatch(PeptideWithSetModifications theBestPeptide, int notch, double score, int scanIndex, Ms2ScanWithSpecificMass scan, DigestionParams digestionParams, List<MatchedFragmentIon> matchedFragmentIons)
-            : base(theBestPeptide, notch, score, scanIndex, scan, digestionParams, matchedFragmentIons)
+        public CrosslinkSpectralMatch(PeptideWithSetModifications theBestPeptide, int notch, double score, int scanIndex, Ms2ScanWithSpecificMass scan, CommonParameters commonParameters, List<MatchedFragmentIon> matchedFragmentIons)
+            : base(theBestPeptide, notch, score, scanIndex, scan, commonParameters, matchedFragmentIons)
         {
-            this.XLTotalScore = score;
+            
         }
 
         public CrosslinkSpectralMatch BetaPeptide { get; set; }
+
         public List<int> LinkPositions { get; set; }
-        public new double DeltaScore { get; set; }
+
         public double XLTotalScore { get; set; } //alpha + beta psmCross.
-        public List<int> XlRank { get; set; } //only contain 2 intger, consider change to Tuple.
-        public string ParentIonExist { get; set; }
-        public int ParentIonExistNum { get; set; }
+        public double SecondBestXlScore { get; set; } // score of the second-best CSM; this is used to calculate delta score
+        public int XlRank { get; set; } //only contain 2 intger, consider change to Tuple.
+
+        //currently unused but will be used again soon for cleavable crosslinker
+        //public string ParentIonExist { get; set; }
+        //public int ParentIonExistNum { get; set; }
+
         public List<int> ParentIonMaxIntensityRanks { get; set; }
         public PsmCrossType CrossType { get; set; }
         public Dictionary<int, List<MatchedFragmentIon>> ChildMatchedFragmentIons { get; set; }
         public int? XlProteinPos { get; private set; }
+
         // loop crosslink protein position 2.
         public int? XlProteinPosLoop { get; private set; }
 
+        //DEBUG don't know if this should be RunnerUpScore or SecondBestXlScore
+        public new double DeltaScore { get { return (XLTotalScore - SecondBestXlScore); } }
```

Comment:
Was this resolved? If so, remove debug comment. If not, resolve it.

Reply:
done

Reply:
This comment is still here?

Reply:
stilll have to push. chill for a sec